tests/schema/tests.py

```diff
@@ -2370,6 +2370,17 @@ def test_add_datefield_and_datetimefield_use_effective_default(self, mocked_date
             cast_function=lambda x: x.time(),
         )
 
+    def test_namespaced_db_table_create_index_name(self):
+        with connection.schema_editor() as editor:
+            max_name_length = connection.ops.max_name_length() or 200
+            namespace = 'n' * max_name_length
+            table_name = 't' * max_name_length
+            index_name = editor._create_index_name(
+                '%s"."%s' % (namespace, table_name), [],
+            )
+        self.assertEqual(index_name[:max_name_length + 3], '%s"."' % namespace)
```

Comment:
I preferred this construct over `self.assertTrue(index_name.startswith('%s"."' % namespace))` because it makes the assertion failure message clearer.